packages/SlideUpDialog/index.vue

```diff
@@ -0,0 +1,157 @@
+<template>
+  <div
+    v-if="value"
+    :class="$style.modal"
+    @click="close"
+    @scroll.self.stop.prevent
+    @wheel.self.stop.prevent
+    @touchmove.self.stop.prevent
+  >
+    <div
+      :class="[$style.container]"
+      @click.stop.prevent
+      @scroll.self.stop.prevent
+      @wheel.self.stop.prevent
+      @touchmove.self.stop.prevent
+    >
+      <!-- @slot modal header -->
+      <slot name="header">
+        <div :class="$style.header" />
+      </slot>
+
+      <div :class="$style.content">
+        <!-- @slot modal content -->
+        <slot />
+      </div>
+
+      <!-- @slot modal footer -->
+      <slot name="footer">
+        <div :class="$style.footer" />
+      </slot>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'SlideUpDialog',
+  props: {
+    /**
+     * show / hide
+     */
+    value: {
+      type: Boolean,
+      default: false
+    },
+    /**
+     * whether allow click outside to close dialog
+     */
+    persistent: {
+      type: Boolean,
+      default: false
+    }
+  },
+  methods: {
+    close() {
+      if (this.persistent) {
+        return false
+      }
+      this.$emit('input', false)
+    }
+  }
+}
+</script>
+
+<style lang="scss" module>
+@import '~/styles/_variables.scss';
+@import '~/styles/extend/flexbox.scss';
+@import '~/styles/extend/card.scss';
+
+.modal {
+  @extend %flex-center;
+  align-items: flex-end;
+
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100vw;
+  height: 100vh;
+  background: rgba($black, 0.55);
+  z-index: 999;
+  animation: motion_bg 0.3s ease-out 0s;
+}
+
+.container {
+  @extend %card;
+  width: 100%;
+  animation: motion_card 0.25s ease-out 0s;
+}
+
+@keyframes motion_bg {
+  0% {
+    background: rgba($black, 0);
+  }
+  100% {
+    background: rgba($black, 0.55);
+  }
+}
+
+@keyframes motion_card {
+  0% {
+    bottom: -80vh;
```

Comment:
Could be larger, since you didn't limit the max-height of `container`

Reply:
I changed it to `-100vh` for now, and I will leave the adjustment to @Bruce-Lan .